src/Dapr.Client/DaprClient.cs

```diff
@@ -163,101 +170,344 @@ public static HttpClient CreateInvokeHttpClient(string appId = null, string dapr
+        /// <param name="httpMethod">The <see cref="HttpMethod" /> to use for the invocation request.</param>
         /// <param name="appId">The Dapr application id to invoke the method on.</param>
-        /// <param name="methodName">The name of the method to invoke.</param>  
-        /// <param name="data">Data to pass to the method</param>      
-        /// <param name="httpOptions">Additional fields that may be needed if the receiving app is listening on HTTP.</param>
+        /// <param name="methodName">The name of the method to invoke.</param>
+        /// <param name="data">The data that will be JSON serialized and provided as the request body.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
-        /// <returns>A <see cref="Task{InvokeResponse}" /> that will return the value when the operation has completed.</returns>
-        public abstract Task<InvocationResponse<TResponse>> InvokeMethodWithResponseAsync<TRequest, TResponse>(
+        /// <returns>A <see cref="Task{T}" /> that will return the value when the operation has completed.</returns>
+        public Task<TResponse> InvokeMethodAsync<TRequest, TResponse>(
+            HttpMethod httpMethod,
             string appId,
             string methodName,
             TRequest data,
-            HttpInvocationOptions httpOptions = default,
-            CancellationToken cancellationToken = default);
+            CancellationToken cancellationToken = default)
+        {
+            var request = CreateInvokeMethodRequest<TRequest>(httpMethod, appId, methodName, data);
+            return InvokeMethodAsync<TResponse>(request, cancellationToken);
+        }
 
         /// <summary>
-        /// Invokes a method on a Dapr app.
+        /// Perform service invocation using gRPC semantics for the application idenfied by <paramref name="appId" /> and invokes the method 
+        /// specified by <paramref name="methodName" /> with an empty request body. 
+        /// If the response has a non-success status code an exception will be thrown.
         /// </summary>
         /// <param name="appId">The Dapr application id to invoke the method on.</param>
-        /// <param name="methodName">The name of the method to invoke.</param>  
-        /// <param name="data">Byte array to pass to the method</param>      
-        /// <param name="httpOptions">Additional fields that may be needed if the receiving app is listening on HTTP.</param>
+        /// <param name="methodName">The name of the method to invoke.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
-        /// <returns>A <see cref="Task{T}" /> that will return the value when the operation has completed.</returns>  
-        public abstract Task<InvocationResponse<byte[]>> InvokeMethodRawAsync(
```

Comment:
Don't we need the raw anymore?

Reply:
We don't need a special method for it anymore - you can use `CreateInvokeMethodRequest()` and then do whatever you like.